pkg/endpoint/policy.go

```diff
@@ -596,11 +617,18 @@ func (e *Endpoint) regeneratePolicy(owner Owner, opts models.ConfigurationMap) (
 // Called with e.Mutex UNlocked
 func (e *Endpoint) regenerate(owner Owner, reason string) (retErr error) {
 	metrics.EndpointCountRegenerating.Inc()
+	regenerateStart := time.Now()
 	defer func() {
 		metrics.EndpointCountRegenerating.Dec()
 		if retErr == nil {
 			metrics.EndpointRegenerationCount.
 				WithLabelValues(metrics.LabelValueOutcomeSuccess).Inc()
+
+			// Capture successful endpoint generation time
+			regenerateTime := time.Since(regenerateStart).Seconds() * secondsToMs
```

Comment:
This must take into account whether regeneration has actually occurred. The majority of regenerations are no-ops because the endpoint does not need to be regenerated. It might even be worth here to count these in the regeneration counter with a special label so we can gain understanding of how many builds are cached.

Reply:
How do we determine that the builds are cached? Can we simply check `compilationExecuted` returned from `revision, compilationExecuted, err := e.regenerateBPF(owner, tmpDir, reason)` know if its cached versus actually regenerated?